### PR TITLE
feat: support api version config

### PR DESCRIPTION
**Pull request by stew-ro**

### CODE REVIEW

**Review by alex-krasn — APPROVED** [review]

File: src/APIConfig.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    "enableAPIVersionSelection" : false
+}
+
+
+
+
+
+
+
+
```
Comment (alex-krasn) [inline src/APIConfig.json:11]: delete empty lines

**Comment by alex-krasn** [comment]
There is an error in console: 
image

**Review by alex-krasn — APPROVED** [review]

File: src/common/buildInfo.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// tslint:disable-next-line:no-var-requires
+const buildConfig = require("../buildConfig.json");
```
Comment (XinxingLiu) [inline src/common/buildInfo.ts:5]: Let's call it something like appConfig, as it applies at runtime instead of build time.

Comment (XinxingLiu) [reply src/common/buildInfo.ts:5]: Also rename buildConfig.json

File: src/common/buildInfo.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Defines the API information
+ */
+export interface IBuildInfo {
```
Comment (XinxingLiu) [inline src/common/buildInfo.ts:10]: Same here.

File: src/react/components/pages/modelCompose/modelCompose.tsx [hunk 4]
```diff
@@ -367,7 +367,7 @@ export default class ModelComposePage extends React.Component<IModelComposePageP
         this.composeModalRef.current.open([], false, false);
 
         if (model.attributes.isComposed) {
-            const apiVersion = this.props.project?.apiVersion || constants.apiVersion;
+            const apiVersion = (constants.enableAPIVersionSelection && this.props.project?.apiVersion) ? this.props.project.apiVersion : constants.apiVersion;
```
Comment (XinxingLiu) [inline src/react/components/pages/modelCompose/modelCompose.tsx:370]: Wrap repeating logic into some helper function?

Comment (XinxingLiu) [inline src/react/components/pages/modelCompose/modelCompose.tsx:370]: Does "this.props.project?.apiVersion" always have value? If its value comes from the apiVersionPicker, it won't be empty, right? In that case we don't need to check  constants.enableAPIVersionSelection either.

Comment (stew-ro) [reply src/react/components/pages/modelCompose/modelCompose.tsx:370]: It **doesn't** always have a value. Older projects created before we added API version selection, have an undefined value. So, it's for backwards compatibility with older projects

**Review by XinxingLiu — APPROVED** [review]

**Review by alex-krasn — APPROVED** [review]

**Review by alex-krasn — APPROVED** [review]

**Review by alex-krasn — APPROVED** [review]

**Pull request merged**